marketplaceItems/nameTag/appResources/appData/resources/modules/entityMaker.js

```diff
@@ -0,0 +1,145 @@
+/*
+
+    Nametag
+    entityMaker.js
+    Created by Milad Nazeri on 2019-02-19
+    Copyright 2019 High Fidelity, Inc.
+
+    Distributed under the Apache License, Version 2.0.
+    See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+
+    A helper library to make entities
+
+*/
+
+
+Script.require('./objectAssign.js');
+
+function EntityMaker(type) {
+    this.properties = {};
+    this.cache = {};
+    this.id = null;
+    this.created = null;
+    this.type = type;
+}
+
+
+// Add properties to the cache / temporary storage
+function add(props){
+    // You can either add an object of props or 2 arguments as key and value
```

Comment:
What's the benefit of having both as an overloaded function with 2 arguments vs 1 argument as a props object?